autogen/main.tf

```diff
@@ -27,7 +27,7 @@ data "google_compute_zones" "available" {
   {% endif %}
 
   project = var.project_id
-  region  = var.region
+  region  = var.region ? var.region : join("-", slice(split("-", var.zones[0]), 0, 2))
```

Comment:
should it be 
`var.region ? var.region : join("-", slice(split("-", var.zones[0]), 0, 2))`
or 
`var.regional ? var.region : join("-", slice(split("-", var.zones[0]), 0, 2))`

Reply:
in first case you check if var region is passed in second if cluster regional

Reply:
In that case I want to make [`region` variable optional for zonal clusters](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/235)

Reply:
I meant in this case if user pass var.regional=true and will not pass var.region, it will have random region right?

Reply:
sorry, not random, but region from zones

Reply:
@kopachevsky in that case we'll got the region name based on the zones, if the `zones` var set. If it's not set - run fails